src/taipy/core/data/parquet.py

```diff
@@ -0,0 +1,222 @@
+    def path(self):
+        return self._path
+
+    @path.setter
+    def path(self, value):
+        self._path = value
+        self.properties[self.__PATH_KEY] = value
+
+    def _check_exposed_type(self, exposed_type):
+        if isinstance(exposed_type, str) and exposed_type not in self.__VALID_STRING_EXPOSED_TYPES:
+            raise InvalidExposedType(
+                f"Invalid string exposed type {exposed_type}. Supported values are {', '.join(self.__VALID_STRING_EXPOSED_TYPES)}"
+            )
+
+    def _read(self):
+        return self.read_with_kwargs()
+
+    def _read_as(self, read_kwargs: Dict):
+        custom_class = self.properties[self.__EXPOSED_TYPE_PROPERTY]
+        list_of_dicts = self._read_as_pandas_dataframe(read_kwargs).to_dict(orient="records")
+        return [custom_class(**dct) for dct in list_of_dicts]
+
+    def _read_as_numpy(self, read_kwargs: Dict):
+        return self._read_as_pandas_dataframe(read_kwargs).to_numpy()
+
+    def _read_as_pandas_dataframe(self, read_kwargs: Dict) -> pd.DataFrame:
+        kwargs = self.properties[self.__READ_KWARGS_PROPERTY]
+        kwargs.update(read_kwargs)
+        return pd.read_parquet(self._path, **kwargs)
+
+    def _read_as_modin_dataframe(self, read_kwargs: Dict) -> modin_pd.DataFrame:
+        kwargs = self.properties[self.__READ_KWARGS_PROPERTY]
+        kwargs.update(read_kwargs)
+        return modin_pd.read_parquet(self._path, **kwargs)
+
+    def _write(self, data: Any, write_kwargs: Dict = dict()):
+        kwargs = {
+            self.__ENGINE_PROPERTY: self.properties[self.__ENGINE_PROPERTY],
+            self.__COMPRESSION_PROPERTY: self.properties[self.__COMPRESSION_PROPERTY],
+        }
+        kwargs.update(self.properties[self.__WRITE_KWARGS_PROPERTY])
+        kwargs.update(write_kwargs)
+        pd.DataFrame(data).to_parquet(self._path, **kwargs)
+
+    def write_with_kwargs(self, data: Any, job_id: Optional[JobId] = None, **kwargs):
+        """Write data, with keyword arguments passed to `pandas.DataFrame.to_parquet`.
+
+        Parameters:
+            data (Any): The data to write.
+            job_id (JobId^): An optional identifier of the writer.
+            **kwargs: The keyword arguments are passed to `pandas.DataFrame.to_parquet`.
+        """
+
+        self._write(data, kwargs)
+        self._last_edit_date = datetime.now()
+        if job_id:
+            self.job_ids.append(job_id)
+
+    def read_with_kwargs(self, **read_kwargs):
+        """Read data node, with keyword arguments passed to `pandas.read_parquet`.
+
+        Parameters:
+            **read_kwargs: The keyword arguments are passed to `pandas.read_parquet`.
+        """
+
+        # return None if data was never written
+        if not self.last_edit_date:
+            self._DataNode__logger.warning(
+                f"Data node {self.id} from config {self.config_id} is being read but has never been " f"written."
+            )
+            return None
+
```

Comment:
```suggestion

        kwargs = self.properties[self.__READ_KWARGS_PROPERTY]
        kwargs.update(read_kwargs)

```
The update is performed here so that we don't have duplication on `. _read_as_pandas_dataframe()` and `._read_as_modin_dataframe()`.

Reply:
Made the changes accordingly, thanks!